### PR TITLE
feat: auto refresh for goBack event in view page [PURCHASE-2788]

### PR DESCRIPTION
**Pull request by alexj105**
The type of this PR is: Feature

This PR resolves [PURCHASE-2788]

### Description
"goBack" listener was added. Also, add this listener for both platforms. 
If I understand correctly `LegacyNativeModules.ARScreenPresenterModule.goBack(unsafe__getSelectedTab())` doesn't provide us refetch logic

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[PURCHASE-2788]: https://artsyproduct.atlassian.net/browse/PURCHASE-2788
ezgif com-gif-maker

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d77c2a80b483d641975a225d0f0e0b2578e3ae69
</p>

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx [hunk 1]
```diff
@@ -33,6 +33,17 @@ const SavedAddresses: React.FC<{ me: SavedAddresses_me; relay: RelayRefetchProp
...
   const addresses = extractNodes(me.addressConnection)
 
+  useEffect(() => {
+    navigationEvents.addListener("goBack", handleModalDismissed)
+    return () => {
+      navigationEvents.removeListener("goBack", handleModalDismissed)
+    }
+  }, [])
+
+  const handleModalDismissed = (): void => {
```
Comment (erikdstock) [inline src/lib/Scenes/SavedAddresses/SavedAddresses.tsx:43]: If the address form is not a modal maybe we could have a different name here?

**Review by erikdstock — COMMENTED** [review]
Looks pretty good to me. 2 comments:
- since there is no modal here in this case maybe `handleModalDismissed` could just be removed in favor of `onRefresh`
- Can we add a test for this?

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx (hunk 1 above)
Comment (alexj105) [reply src/lib/Scenes/SavedAddresses/SavedAddresses.tsx]: Good notes. It was solved, ready to review again :)

File: src/lib/Scenes/SavedAddresses/__tests__/SavedAddresses-tests.tsx [hunk 2]
```diff
@@ -126,4 +126,11 @@ describe(SavedAddressesQueryRenderer, () => {
     tree.findByType(Button).props.onPress()
     expect(navigate).toHaveBeenCalledWith("/my-profile/saved-addresses/new-address")
   })
+
+  it("handles return to prev view with goBack event", () => {
+    const mockCallback = jest.fn(util.onRefresh)
+    navigationEvents.addListener("goBack", mockCallback)
+    navigationEvents.emit("goBack")
+    expect(mockCallback.mock.calls.length).toBe(1)
```
Comment (erikdstock) [inline src/lib/Scenes/SavedAddresses/__tests__/SavedAddresses-tests.tsx:134]: Looks good. I was trying to find an example of asserting something like `expect(relay.refetch).toHaveBeenCalled()` where relay has been mocked. @lilyfromseattle pointed me to [Conversation-tests.tsx](https://github.com/artsy/eigen/blob/dd84162a6c66f9cd552da09842ce119098eef918/src/lib/Scenes/Inbox/Screens/__tests__/Conversation-tests.tsx#L99) but we realized this is a case using a class component. Not sure if we have a good pattern for this at the moment- let's merge and consider revisiting later.

**Review by erikdstock — APPROVED** [review]

File: src/lib/Scenes/SavedAddresses/__tests__/SavedAddresses-tests.tsx [hunk 3]
```diff
@@ -133,4 +133,20 @@ describe(SavedAddressesQueryRenderer, () => {
...
+      name: "saver",
+      addressConnection: {
+        edges: [],
+      },
+    }
+    const tree = renderWithWrappers(<SavedAddresses me={mockMe as any} relay={relayMock} />).root
+
+    tree.findByType(Button).props.onPress()
+    expect(navigate).toHaveBeenCalledWith("/my-profile/saved-addresses/new-address") // works
+    navigationEvents.emit("goBack")
```
Comment (erikdstock) [inline src/lib/Scenes/SavedAddresses/__tests__/SavedAddresses-tests.tsx:149]: I've poked at this but cannot figure out why this call isn't being picked up by the rendered component. I've also tried it with the normal `<TestRenderer />` above. Maybe something about it being a function component? If we don't have a solution by tomorrow I say let's merge!

**Pull request merged**